app/Http/Middleware/AuthApi.php

```diff
@@ -6,35 +6,74 @@
 namespace App\Http\Middleware;
 
 use Closure;
-use Illuminate\Auth\Middleware\Authenticate;
-use Illuminate\Http\Request;
+use Illuminate\Auth\AuthenticationException;
+use Laravel\Passport\TokenRepository;
+use League\OAuth2\Server\Exception\OAuthServerException;
+use League\OAuth2\Server\ResourceServer;
+use Symfony\Bridge\PsrHttpMessage\Factory\PsrHttpFactory;
+use Zend\Diactoros\ResponseFactory;
+use Zend\Diactoros\ServerRequestFactory;
+use Zend\Diactoros\StreamFactory;
+use Zend\Diactoros\UploadedFileFactory;
 
 class AuthApi
 {
-    const SKIP_GET = [
-        'api/v2/changelog/',
-        'api/v2/comments/',
-        'api/v2/seasonal-backgrounds/',
-        'api/v2/wiki/',
-    ];
-
-    // TODO: this should be definable per-controller or action.
-    public static function skipAuth($request)
-    {
-        $path = "{$request->decodedPath()}/";
+    const REQUEST_OAUTH_TOKEN_KEY = 'oauth_token';
+
+    protected $server;
+    protected $repository;
 
-        return $request->isMethod('GET') && starts_with($path, static::SKIP_GET);
+    public function __construct(ResourceServer $server, TokenRepository $repository)
+    {
+        $this->server = $server;
+        $this->repository = $repository;
     }
 
-    public function handle(Request $request, Closure $next)
+    public function handle($request, Closure $next)
     {
         auth()->shouldUse('api');
```

Comment:
I haven't checked deeper but I suspect you can remove this and set user manually when validating token instead. Something like this: https://0paste.com/72919-e7aaf5a7.txt

Although even then this still does query same token twice similar to https://github.com/laravel/passport/issues/382 but in this class' `validTokenFromRequest` instead (in addition to oauth library's).

Reply:
at over 600 queries for beatmapsets search endpoint using client cred, I think this needs to be fixed

![](https://s.myconan.net/2020-07/firefox_2020-07-06_17-47-45.png)